### ISSUE TITLE
405GAL doesn't seem to take into account variables for template src

### ISSUE THREAD

**Issue by geerlingguy** [ansible/ansible-lint#385]
> Note: This was copied over from the [galaxy-lint-rules] repository; please feel free to close if it doesn't currently apply to the existing or planned set of ansible-lint rules. Original issue: https://github.com/ansible/galaxy-lint-rules/issues/13

For my GitLab role, I have the following task:

```yaml
- name: Copy GitLab configuration file.
  template:
    src: "{{ gitlab_config_template }}"
    dest: /etc/gitlab/gitlab.rb
    owner: root
    group: root
    mode: 0600
  notify: restart gitlab
```

And `gitlab_config_template` is defined as:

```yaml
gitlab_config_template: "gitlab.rb.j2"
```

When I run Ansible Lint on it with these rules I get:

```
[405GAL] Template files should have the extension '.j2' 
/Users/jeff.geerling/Dropbox/VMs/roles/geerlingguy.gitlab/tasks/main.yml:69
Task/Handler: Copy GitLab configuration file.
```

It looks like 405GAL is not taking into account variable expansion for template names—not sure if this is possible to do, but there are a number of roles where I use a variable for the template to allow role users to override the template location.

(Also, aside: it seems like the output for this rule has an extra space on the end `[405GAL] Template files should have the extension '.j2' ` (hard to convey without an empty whitespace highlighter, but it's there when I copy and paste into my IDE, whereas it's not there for any other rules)).

**Comment by geerlingguy** [comment]
Possibly related to: https://github.com/willthames/ansible-lint/pull/384

**Comment by awcrosby** [comment]
for #384 - [ModuleTemplateExt 405](https://github.com/willthames/ansible-lint/pull/384/files#diff-de8f1ae22c1f730dce47f9c58339bb72)... I'm not sure if the variable expansion is possible, but for now we can check if variable format exists and if so exit without flagging a linter issue.

**Comment by geerlingguy** [comment]
@awcrosby - That would work for me.
